quantum/pointing_device/pointing_device_modes.c

```diff
@@ -0,0 +1,640 @@
+        return INT16_MIN;
+    } else if (value > INT16_MAX) {
+        return INT16_MAX;
+    } else {
+        return value;
+    }
+}
+
+static inline uint16_t clamp_uint_16_to_8(uint16_t value) {
+    if (value > UINT8_MAX) return UINT8_MAX;
+    return value;
+}
+
+static int16_t divisor_multiply16(int16_t value) {
+    return clamp_int_32_to_16(value * (int16_t)current_divisor);
+}
+
+static int8_t divisor_divide8(int16_t value) {
+    return clamp_int_16_to_8(value / (int16_t)current_divisor);
+}
+
+static int16_t divisor_divide16(int16_t value) {
+    return value / (int16_t)current_divisor;
+}
+
+/**
+ * @brief divides value by current divisor return mouse x/y
+ *
+ * Intended for modifying pointing_mode x/y values
+ *
+ * @params value[in] value int16_t
+ *
+ * @return quotient clamped to mouse_xy_report_t range
+ */
+mouse_xy_report_t apply_divisor_xy(int16_t value) {
+#    ifdef MOUSE_EXTENDED_REPORT
+    return divisor_divide16(value);
+#    else
+    return divisor_divide8(value);
+#    endif
+}
+
+/**
+ * @brief divides value by current divisor return mouse h/v
+ *
+ * Intended for modifying pointing_mode x/y values
+ *
+ * @params value[in] value int16_t
+ *
+ * @return quotient clamped to int8_t range
+ */
+int8_t apply_divisor_hv(int16_t value) {
+    return divisor_divide8(value);
+}
+
+/**
+ * @brief multiplies mouse x/y value by current divisor
+ *
+ * Intended for collecting residuals by subtracting quotient * divisor from the dividend
+ * as modulo can be large and slow on some chips
+ *
+ * @params value[in] value mouse_xy_report_t mouse x/y axis value to multiply
+ *
+ * @return product clamped to int16_t range
+ */
+int16_t multiply_divisor_xy(mouse_xy_report_t value) {
+    return divisor_multiply16((int16_t)value);
+}
+
+/**
+ * @brief multiplies mouse h/v value by current divisor
+ *
+ * Intended for collecting residuals by subtracting quotient * divisor from the dividend
+ * as modulo can be large and slow on some chips
+ *
+ * @params value[in] value int8_t mouse h/v axis value to multiply
+ *
+ * @return modified and clamped value int16_t
+ */
+int16_t multiply_divisor_hv(int8_t value) {
+    return divisor_multiply16((int16_t)value);
+}
+
+/**
+ * @brief Return device id of current controlled device
+ *
+ * @return current device id [uint8_t]
+ */
+uint8_t get_pointing_mode_device(void) {
+#    ifdef POINTING_MODES_SINGLE_CONTROL
+    return selected_device;
+#    else
+    return current_device;
+#    endif
+}
+
+/**
+ * @brief Allow changing of active side
+ *
+ * will change which side (PM_LEFT_DEVICE, PM_RIGHT_DEVICE, etc.) is controlled by pointing mode framework
+ *
+ * NOTE: If mode is set above maximum device number device is set to zero (this allows cycling)
+ *
+ * @params[in] new side uint8_t
+ */
+void set_pointing_mode_device(uint8_t device) {
+#    if (POINTING_MODES_NUM_DEVICES > 1)
+    if (device > POINTING_MODES_NUM_DEVICES) device = 0;
+#        if POINTING_MODES_SINGLE_CONTROL
```

Comment:
This errors out for me. I might be wrong but it compiled when I changed it to` #        if defined(POINTING_MODES_SINGLE_CONTROL)`

Reply:
Yeah I see the mistake I made here, so I'll fix that up and make sure I can compile it.